File: src/main/java/me/schlaubi/votebot/commands/vote/ChangeHeadingCommand.java
package me.schlaubi.votebot.commands.vote;

import me.schlaubi.votebot.core.VoteManager;
import me.schlaubi.votebot.core.command.Command;
import me.schlaubi.votebot.core.command.CommandCategory;
import me.schlaubi.votebot.core.command.CommandEvent;
import me.schlaubi.votebot.core.command.Result;
import me.schlaubi.votebot.core.command.permission.Permissions;
import me.schlaubi.votebot.core.entities.Vote;

public class ChangeHeadingCommand extends Command {

    public ChangeHeadingCommand() {
        super(new String[] {"changeheading"}, CommandCategory.VOTE, Permissions.everyone(), "Let's you change the heading of a poll", "<heading>");
    }

    @Override
    public Result run(String[] args, CommandEvent event) {
        if (args.length == 0)
            return sendHelp();
        VoteManager manager = event.getBot().getVoteManager();
        if (!manager.hasVote(event.getMember()))
            return send(error(event.translate("command.close.novote.title"), event.translate("command.close.novote.description")));
        Vote vote = manager.getVote(event.getMember());
        String heading = event.getArguments();
        if (heading.equals(vote.getHeading()))
            return send(error(event.translate("command.changeheading.duplication.title"), event.translate("command.changeheading.duplication.description")));
        vote.changeHeading(heading);
        vote.updateMessages();
        return send(success(event.translate("command.changeheading.success.title"), String.format(event.translate("command.changeheading.success.description"), heading)));
    }
}

File: src/main/kotlin/me/schlaubi/votebot/commands/vote/ChangeHeadingCommand.kt
/*
 * VoteBot - A unique Discord bot for surveys
 *
 * Copyright (C) 2019  Michael Rittmeister
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see https://www.gnu.org/licenses/.
 */

package me.schlaubi.votebot.commands.vote

import cc.hawkbot.regnum.client.command.Group
import cc.hawkbot.regnum.client.command.context.Arguments
import cc.hawkbot.regnum.client.command.context.Context
import cc.hawkbot.regnum.client.command.permission.CommandPermissions
import cc.hawkbot.regnum.client.util.EmbedUtil
import me.schlaubi.votebot.VOTE
import me.schlaubi.votebot.commands.VoteBotCommand
import me.schlaubi.votebot.core.VoteBot

class ChangeHeadingCommand(
    bot: VoteBot
) : VoteBotCommand(
    bot,
    Group.VOTE,
    "Change heading",
    arrayOf("changeheading", "heading", "ch"),
    CommandPermissions(public = true, node = "changeheading"),
    "<newHeading>",
    "Is this a useless heading?",
    "Updates the heading of a Vote"
) {
    override fun execute(args: Arguments, context: Context) {
        if (args.isEmpty()) {
            return context.sendHelp().queue()
        }
        hasVote(context) {
            val heading = args.string<String>()
            it.controller.setHeading(heading)
            context.sendMessage(
                EmbedUtil.success(
                    context.translate("command.changeheading.success.title"),
                    context.translate("command.changeheading.success.description")
                        .format(heading)
                )
            ).queue()
        }
    }
}